wsmaster/codenvy-hosted-api-permission/src/main/java/com/codenvy/api/permission/server/PermissionsManager.java

```diff
@@ -73,6 +77,7 @@ public PermissionsManager(EventService eventService,
         }
         this.domains = ImmutableList.copyOf(domains);
         this.domainToDao = ImmutableMap.copyOf(domainToDao);
+        this.updateLocks = new StripedLocks(16);
```

Comment:
Single lock instance will lock updating of all permissions.
StripedLocks distribute locking of permissions update by instance id.

Do you think it would be better to lock all permissions here?